external/llvm-project/mlir/lib/Conversion/GPUToROCDL/LowerGpuOpsToROCDLOps.cpp

```diff
@@ -148,6 +150,15 @@ struct LowerGpuOpsToROCDLOpsPass
     if (failed(applyPartialConversion(m, target, std::move(llvmPatterns))))
       signalPassFailure();
 
+    // Manually rewrite known block size attributes so the LLVMIR translation
+    // infrastructure can pick them up.
+    m.walk([](LLVM::LLVMFuncOp op) {
+      if (Attribute blockSizes =
+              op->removeAttr(gpu::GPUFuncOp::getKnownBlockSizeAttrName()))
+        op->setAttr(ROCDL::ROCDLDialect::getReqdWorkGroupSizeAttrName(),
```

Comment:
So `KnownBlockSize` and `ReqdWorkGroupSize` are equivalent attributes but used across two abstraction layers? Why bother with the rename?

Reply:
The mlir-to-llvmir translator won't let the ROCDL translator use an attribute to modify a function unless it starts with `rocdl.`, so we have to do some sort of rename. This also makes it clearer what the attribute corresponds to in the LLVM IR, which uses `!reqd_work_group_size` for this annotation.